### PR TITLE
feat(FX-3768): Add additional event tracking

### PR DESCRIPTION
**Pull request by dimatretyak**
Type: **Enhancement**
Jira ticket: [FX-3768]

### Acceptance Criteria
* Add `editedSavedSearch` event when the saved search alert was updated
* Track `deletedSavedSearch` event when the saved search alert was deleted

### Useful Links
* [Find and Explore Analytics Tracking](https://docs.google.com/spreadsheets/d/1_VDAmRJz2K87FXDq7oh_tQSgwBv4syuZjaaR3sfetcI/edit#gid=121201130)
* [Notion card](https://www.notion.so/artsy/We-d-need-2-additional-event-tracking-179d8c06383e4a9eafe86f6c7eb8b466)

#### editedSavedSearch event
https://user-images.githubusercontent.com/3513494/155741736-4785f666-d52e-43b7-9740-d76e25ed8ef5.mp4

updated-event

#### deletedSavedSearch event
https://user-images.githubusercontent.com/3513494/155741788-3bfd2df2-e903-485c-9d98-f1df2f1d5387.mp4

delete-event

[FX-3768]: https://artsyproduct.atlassian.net/browse/FX-3768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dblandin — APPROVED** [review]
Looks great 👍🏼

File: src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/__tests__/SavedSearchAlertEditForm.jest.tsx [hunk 1]
```diff
@@ -120,6 +129,42 @@ describe("SavedSearchAlertEditForm", () => {
...
+        name: "Alert #1",
+        email: true,
+        push: false,
+      }),
+      changed: JSON.stringify({
+        name: "Updated Name",
+        push: false,
+        email: true,
+      }),
+    })
```
Comment (gkartalis) [inline src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/__tests__/SavedSearchAlertEditForm.jest.tsx:165]: **suggestion(non-blocking):** what do you think of using `toMatchInlineSnapshot` here as we do in other analytics assertions? (like [this one](https://github.com/artsy/force/blob/c28fc18c856513a7d201886513c80e213ff916da/src/v2/Components/Authentication/__tests__/FormSwitcher.jest.tsx#L138-L152))

**Review by gkartalis — DISMISSED** [review]
Left a tiny comment but looking great! 🚀

File: src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/__tests__/SavedSearchAlertEditForm.jest.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/__tests__/SavedSearchAlertEditForm.jest.tsx:165]: Done

**Pull request merged**